pkgs/stdenv/generic/setup.sh

```diff
@@ -445,7 +445,7 @@ substituteAll() {
 
     # Select all environment variables that start with a lowercase character.
     # Will not work with nix attribute names (and thus env variables) containing '\n'.
```

Comment:
I saw the comment... and that it only referred to names. I don't know of any occurrence with a newline in variable name, but we do have lots of newlines in values, as we pass phases as variables.

Personally, I find it rather hard to program fully correctly in bash. Everything is just text even if it means something else, and unwanted expansions get into my way all the time. (Still, on typical inputs the scripts work somehow.)